src/EnergyPlus/PlantLoopEquip.cc

```diff
@@ -699,10 +668,12 @@ namespace PlantLoopEquip {
             // for heat recovery plant interactions.
 
             if (EquipTypeNum == TypeOf_Generator_FCExhaust) {
+                // TODO: Need to set this during GetInput, not here
```

Comment:
@mitchute would you want to take a crack at cleaning this up?  I'm really just trying to make every single component uniform here before we strip out this entire IF block structure.  I need these dynamic casts gone.  Can these TypeOf just be set on the components themselves during their GetInput routines?  If not there, then could they be set once at the call to the factory in PlantManager?  I see these two components share the same data structure but they have different EquipTypeNum so I know it may be a little tricky, but it seems the plant solver definitely already knows the difference here.

Reply:
You mean like the one I added over on the ElectricEIR refactor: https://github.com/NREL/EnergyPlus/pull/7679/files#diff-ae587fad7fd53d7ab16d3d93e273bcefR323
😄 

I'll take a crack at it.

Reply:
:smile: well the EquipCtrlType is pretty straightforward -- once the component calls ScanPlantLoopsForObject that assignment can be made from knowledge of the loop/side/br/comp right there.  In this case, it's a little less obvious, but hopefully still pretty easy.